### ISSUE TITLE
Retry connection connection closed during resolution.

### ISSUE THREAD

**Issue by nikron** [gradle/gradle#16642]
Currently, some error types are marked as [network issues](https://github.com/gradle/gradle/blob/master/subprojects/dependency-management/src/main/java/org/gradle/api/internal/artifacts/repositories/transport/NetworkingIssueVerifier.java). I'd like to include `org.apache.http.ConnectionClosedException`.

### Expected Behavior
org.apache.http.ConnectionClosedException retries

### Current Behavior
org.apache.http.ConnectionClosedException fails

### Context
We have some rather large jars that need to get downloaded. We have a lot of CI builds. Very occasionally, the jar download fails in the middle, causing an unnecessary build failure.

**Comment by nikron** [comment]
For what it's worth, I'm willing to create this pull request if this change is likely to be accepted.

**Comment by ljacomet** [comment]
I believe this makes sense, @melix do you agree?

Based on the answer, please go ahead with a PR.

**Comment by melix** [comment]
Agreed. Although I'm surprised we would surface an `org.apache.http` exception here, but my memory may not be correct.

**Comment by nikron** [comment]
I'm not sure what you mean by "surface"; it's part of the cause chain, so it's not the top level cause. `NetworkingIssueVerifier` already recursively explores the cause chain.
